entrypoint.sh

```diff
@@ -6,6 +6,11 @@ sed -i s/localhost/$HOSTNAME/g /etc/slurm-llnl/slurm.conf
 /etc/init.d/slurmctld start
 /etc/init.d/slurmd start
 #/etc/init.d/postgres start
+if [ $(psql -Atc 'select count(distinct table_schema) from information_schema.tables;') -eq 2 ]; then
+  psql -c "CREATE USER web_usr PASSWORD 'postgres';"
+  psql -f t/data/fixture/empty_fixture.sql
+  ( cd db && ./run_all_patches.pl -u ${PGUSER} -p "${PGPASSWORD}" -h ${PGHOST} -d ${PGDATABASE} -e janedoe )
```

Comment:
This approach requires setting these environment variables somewhere. A few options:

1. Hard-cod in the docker-compose.yml:
```
services:
  breedbase:
    ...
    environment:
      PGDATABASE: breedbase
      PGHOST: breedbase_db
      PGPASSWORD: postgres
      PGUSER: postgres
    ...
```

drawback: it might not be considered a "best practice" from a security perspective to put secrets in a config file that presumably users will want to version control.

2. Require the environment variables be set in the environment of the `docker-compose` command on the host; e.g., the user can create a [.env file](https://docs.docker.com/compose/environment-variables/#the-env-file) in the same directory as docker-compose.yml, such as:

```
PGDATABASE=breedbase
PGHOST=breedbase_db
PGPASSWORD=postgres
PGUSER=postgres
```

Then docker-compose.yml can set the corresponding environment variables in the service from these environment variables:

```
services:
  breedbase:
    ...
    environment:
      PGDATABASE: ${PGDATABASE:?}
      PGHOST: ${PGHOST:?}
      PGPASSWORD: ${PGPASSWORD:?}
      PGUSER: ${PGUSER:?}
    ...
```

and if a user forgets to set the environment variables (e.g., in a .env file), an error results:
```
$ docker-compose up -d
ERROR: Missing mandatory value for "environment" option interpolating {'PGDATABASE': '${PGDATABASE:?}', 'PGHOST': '${PGHOST:?}', 'PGPASSWORD': '${PGPASSWORD:?}', 'PGUSER': '${PGUSER:?}'} in service "breedbase": 
```

3. Require a .pgpass file be bind-mounted to $HOME/.pgpass in the breedbase service. It seems this would at least require run_all_patches.pl to support being run without specifying `--user`, `--pass`, etc.

4. Use [peer authentication](https://github.com/docker-library/postgres/pull/778#issuecomment-717594693) to obviate the need for passwords: the www-data user automatically authenticates as web_usr. This requires the postgres Unix domain socket to be accessible to the breedbase service, which can be accomplished using a Docker volume. It would also require the initial `psql` commands in entrypoint.sh execute as the 'postgres' user---however, since the postgres user doesn't exist in the breedbase_web container, something like running `psql` as the UID of the postgres user in the postgres container is necessary; e.g.:


```
index 527119a..23107f3 100644
--- a/docker-compose.yml
+++ b/docker-compose.yml
@@ -2,6 +2,12 @@ version: "3.7"
 
 services:
   breedbase:
+    environment:
+      PGHOST: /run/postgresql
+      PGPORT:
+      PGPASSWORD:
+      PGUSER:
+      PGDATABASE: breedbase
     volumes:
       - type: bind
         source: ./sgn_local.conf
@@ -15,6 +21,7 @@ services:
       - cluster:/home/production/cluster
       - blast:/home/production/blast
       - images:/home/production/images
+      - postgres-socket:/run/postgresql
     ports:
       - 7080:8080
       - 25:25
@@ -26,13 +33,15 @@ services:
   breedbase_db:
     volumes:
       - dbdata:/var/lib/postgresql/data
+      - postgres-socket:/run/postgresql
     image: postgres:12.6
     container_name: breedbase_db
     environment:
       POSTGRES_DB: breedbase
-      POSTGRES_PASSWORD: postgres
+      POSTGRES_INITDB_ARGS: --auth=peer
+      POSTGRES_PASSWORD: unused
     healthcheck:
-      test: "pg_isready -U postgres -d $${POSTGRES_DB} || exit 1"
+      test: "sudo -u postgres pg_isready -d $${POSTGRES_DB} || exit 1"
       interval: 10s
 
 volumes:
@@ -43,3 +52,4 @@ volumes:
   blast:
   images:
   dbdata:
+  postgres-socket:
diff --git a/entrypoint.sh b/entrypoint.sh
index 1d4dfdb..403b59e 100644
--- a/entrypoint.sh
+++ b/entrypoint.sh
@@ -6,10 +6,10 @@ sed -i s/localhost/$HOSTNAME/g /etc/slurm-llnl/slurm.conf
 /etc/init.d/slurmctld start
 /etc/init.d/slurmd start
 #/etc/init.d/postgres start
-if [ $(psql -Atc 'select count(distinct table_schema) from information_schema.tables;') -eq 2 ]; then
-  psql -c "CREATE USER web_usr PASSWORD 'postgres';"
-  psql -f t/data/fixture/empty_fixture.sql
-  ( cd db && ./run_all_patches.pl -u ${PGUSER} -p "${PGPASSWORD}" -h ${PGHOST} -d ${PGDATABASE} -e janedoe )
+if [ $(sudo -u '#999' psql -U postgres -Atc 'select count(distinct table_schema) from information_schema.tables;') -eq 2 ]; then
+  sudo -u '#999' psql -U postgres -c 'CREATE USER web_usr; CREATE ROLE "www-data" IN ROLE web_usr;'
+  sudo -u '#999' psql -U postgres -f t/data/fixture/empty_fixture.sql
+  ( cd db && sudo -u '#999' ./run_all_patches.pl -u postgres -p '' -h "${PGHOST}" -d "${PGDATABASE}" -e janedoe )
 fi
 
 if [ "$MODE" == "DEVELOPMENT" ]; then
```